testing/src/main/scala/org/http4s/testing/ArbitraryInstances.scala

```diff
@@ -42,6 +42,36 @@ trait ArbitraryInstances {
       } yield NonEmptyList(a, list)
     }
 
+  val genChar: Gen[Char] = choose('\u0000', '\u007F')
+
+  val ctlChar: List[Char] = ('\u007F' +: ('\u0000' to '\u001F')).toList
+
+  val lws: List[Char] = " \t".toList
+
+  val genCrLf: Gen[String] = const("\r\n")
+
+  val genRightLws: Gen[String] = nonEmptyListOf(oneOf(lws)).map(_.mkString)
+
+  val genLws: Gen[String] =
+    oneOf(sequence[List[String], String](List(genCrLf, genRightLws)).map(_.mkString), genRightLws)
+
+  val octets: List[Char] = ('\u0000' to '\u00FF').toList
+
+  val genOctet: Gen[Char] = oneOf(octets)
+
+  val allowedText: List[Char] = octets.diff(ctlChar)
+
+  val genText: Gen[String] = oneOf(nonEmptyListOf(oneOf(allowedText)).map(_.mkString), genLws)
+
+  val allowedQDText: List[Char] = allowedText.filterNot(c => c == '"' || c == '\\')
```

Comment:
According to spec the backslash character `\` is allowed but it seems the parser cannot handle one particular case where the Generated QDText ends on `\"` like e.g.

`"_¡1S^¼m¬Á\"`
`"ìÙ\"`

This is certainly a border case and I couldn't figure how to make the parboiled parser handle it properly so I tweaked the generator. WDYT?

Reply:
Certainly seems like we _should_ be able to fix the parser.  I don't want it to stand in the way of progress, but I'd like to leave a note so we try to fix it.